src/main/BlockPool.cpp

```diff
@@ -0,0 +1,1216 @@
+/*
```

Comment:
This file has gotten quite large and complex.  Please split it into multiple files.

Allocator.*
AllocatorSuperblock.*
AllocatorTable.*
etc.

Thanks